examples/speculative/speculative.cpp

```diff
@@ -154,6 +159,14 @@ int main(int argc, char ** argv) {
             break;
         }
 
+        if (drafted.size() > 0 && all_accepted) {
+            n_draft += 2;
+            LOG("all drafted tokens accepted, n_draft = %d\n", n_draft);
+        } else {
+            n_draft -= 1;
+            LOG("drafted token rejected, n_draft = %d\n", n_draft);
+        }
```

Comment:
Why does `n_draft` go up by `2` when all drafted tokens are accepted, but decrease by `1` when a drafted token is rejected? Is there a more efficient algorithm to handle this? The current approach seems similar to TCP Friendly Rate Control.

Reply:
That's basically copied from huggingface's implementation, we may do some param searchning to optimize that.